:recycle: restructure addition docs

eve-bright opened the pull request:
----

If you open a PR that needs to go into a current version, you need to *cherry-pick your commit from dev over to the current version branch*. Only then will the proper builds that generate html/pdf be run. But beware: Docs will be generated but not published automatically!

- [x] N/A - or - I have added the appropriate "cherry-pick-to" labels to this PR so I don't forget to do this later!

eve-bright committed:
:recycle: restructure addition docs

neo-technology-commit-status-publisher commented:
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.
<!-- Sticky Pull Request Comment -->

ravikiran-sulikeri commented on modules/ROOT/pages/addition/agent-installation/self-registered.adoc:
```diff
@@ -1,55 +1,179 @@
++
+image::agent-approve.png[width=800]
+.. Upon approval, the agent status changes to `Offline` until the agent receives token information and re-connects to NOM server.
+image::agent-approved-offline.png[width=800]
```
this image is not rendered

eve-bright replied:
Oops - good catch!

eve-bright committed:
fix image for approved agent

alexnb committed:
there is no cog icon anymore, I think it was replaced with the table row's menu

alexnb commented on modules/ROOT/pages/addition/agent-installation/self-registered.adoc:
```diff
@@ -1,55 +1,180 @@
+----
+[Service]
+Environment="CONFIG_SERVER_ADDRESS=<server grpc address>"
+Environment="CONFIG_TOKEN_URL=<server http login url>"
+Environment="CONFIG_TOKEN_CLIENT_ID=<client id>"
+Environment="CONFIG_TOKEN_CLIENT_SECRET=<client secret>"
```
```suggestion
```

Since it is self-registering, we don't need `CONFIG_TOKEN_*`?

alexnb commented on modules/ROOT/pages/addition/agent-installation/self-registered.adoc:
```diff
@@ -1,55 +1,180 @@
 ----
-<agent name> <agent version> <agent revision>
+CONFIG_SERVER_ADDRESS=<server grpc address>
+CONFIG_TOKEN_URL=<server http login url>
+CONFIG_TOKEN_CLIENT_ID=<client id>
+CONFIG_TOKEN_CLIENT_SECRET=<client secret>
```
```suggestion
```

Since it is self-registering, we don't need `CONFIG_TOKEN_*`?

alexnb committed:
just typos

alexnb committed:
repaired the link

eve-bright committed:
Fix example service variable definitions

eve-bright commented on modules/ROOT/pages/addition/agent-installation/manual.adoc:
```diff
@@ -175,7 +298,7 @@ The rest of NOM still works.
 
 | `CONFIG_INSTANCE_n_LOG_CONFIG_PATH`
-| Path to the instance <<server_log_config,log4j config file>>.
+| Path to the instance xref:querylog[log4j config file].
```
I missed this completely 🤦. 

Do you think it would be even better if we re-instated a separate section for the location of that file? Maybe as a subsection of the querylog section?

alexnb replied:
Yes, also to reduce duplication, we should do something here.

I suggest putting the sentence " If CONFIG_INSTANCE_n_QUERY_LOG_PORT is not set, the feature is treated as disabled. The rest of NOM still works." to the description of CONFIG_INSTANCE_n_QUERY_LOG_PORT.

Then move "The location of server log configuration ..." including the IMPORTANT section to "Query log collection" on the page "Neo4j instance requirements" including adjusting the links from CONFIG_INSTANCE_n_LOG_CONFIG_PATH to point there.

WDYT?

eve-bright replied:
That sounds like a good plan 👍

alexnb replied:
OK, so I'll commit it to your branch so that you can review.

alexnb committed:
just typos

alexnb committed:
fixed broken link

alexnb committed:
Put the sentence " If CONFIG_INSTANCE_n_QUERY_LOG_PORT is not set, the feature is treated as disabled. The rest of NOM still works." to the description of CONFIG_INSTANCE_n_QUERY_LOG_PORT.

Then moved "The location of server log configuration ..." including the IMPORTANT section to "Query log collection" on the page "Neo4j instance requirements" including adjusting the links from CONFIG_INSTANCE_n_LOG_CONFIG_PATH to point there.

alexnb committed:
it should be clear that the rest would work

alexnb reviewed (approved):
Please see my commits on moving the query log section.

eve-bright commented:
> Please see my commits on moving the query log section.

LGTM!

AlexicaWright commented on modules/ROOT/pages/addition/agent-installation/agent-install.adoc:
```diff
@@ -0,0 +1,36 @@
+
+* Download platform specific binary: https://neo4j.com/download-center/#ops-manager[Download]
++
+NOTE: If downloading agents is not a option in some environments, NOM agent binaries are also bundled with Neo4j packages.
+NOM agent packages can be found in the `products` folder in the Neo4j home directory. 
+These binaries have support for the Neo4j version that they are bundled with.
```
```suggestion
[NOTE]
==== 
In certain environments, downloading agents is not an option.
Therefore, NOM agent binaries are also bundled with Neo4j packages.
NOM agent packages can be found in the `products` folder in the Neo4j home directory. 
These binaries have support for the Neo4j version that they are bundled with.
====
```

AlexicaWright commented on modules/ROOT/pages/addition/agent-installation/manual.adoc:
```diff
@@ -31,58 +35,156 @@ You can copy them over to the host for ease of agent configuration.
+    It may be that the server address is configured incorrectly or the TLS certificates are not correctly specified. 
+. Verify that the agent has a current status of `Online`. 
+    ** If the agent is not currently online then go to where it is running and check the logs.
+. Hover over the newly added agent and select "View Configuration" from the menu on the right to show agent configuration. Check configuration is as expected.
```
```suggestion
. Hover over the newly added agent and select "View Configuration" from the menu on the right to show agent configuration. 
Check configuration is as expected.
```

AlexicaWright commented on modules/ROOT/pages/addition/agent-installation/manual.adoc:
```diff
@@ -105,17 +207,28 @@ NOTE: For a self registering agent, token configuration below is not a requireme
+====
+Since agent-server communication needs to be encrypted, you need to configure the agent so that it trusts the server's certificates.
+The file that contains the trusted certificate list (PEM encoded) can be specified through the `CONFIG_TLS_TRUSTED_CERTS` environment variable.
+While most operating systems default to the system-wide trusted certificates, that's not the case on Windows.
```
```suggestion
Most operating systems default to the system-wide trusted certificates, but that is not the case on Windows.
```

AlexicaWright commented on modules/ROOT/pages/addition/agent-installation/self-registered.adoc:
```diff
@@ -1,55 +1,176 @@
 
-* Make sure the agent version is equal to that of the NOM server.
+[[verify]]
+== Verify agent setup
```
```suggestion
== Verify agent setup

```

AlexicaWright commented on modules/ROOT/pages/addition/agent-installation/self-registered.adoc:
```diff
@@ -1,55 +1,176 @@
++
+image::agent-approved-online.png[width=800]
+. If the agent status is not 'Online' then go to where it is running and check the logs.
+. Hover over the newly added agent and select "View Configuration" from the menu on the right to show agent configuration. Check configuration is as expected.
```
```suggestion
. Hover over the newly added agent and select "View Configuration" from the menu on the right to show agent configuration. 
Check configuration is as expected.
```

AlexicaWright commented on modules/ROOT/pages/addition/agent-installation/self-registered.adoc:
```diff
@@ -64,10 +185,21 @@ Which should output:
+====
+Since agent-server communication needs to be encrypted, you need to configure the agent so that it trusts the server's certificates.
+The file that contains the trusted certificate list (PEM encoded) can be specified through the `CONFIG_TLS_TRUSTED_CERTS` environment variable.
+While most operating systems default to the system-wide trusted certificates, it is not the case on Windows.
```
```suggestion
Most operating systems default to the system-wide trusted certificates, but this is not the case on Windows.
```

AlexicaWright commented on modules/ROOT/pages/addition/docker/index.adoc:
```diff
@@ -1,7 +1,7 @@
 In this section we will list some of the methods to setup agents to monitor containerized Neo4j instances in standalone and Docker compose deployments. For Kubernetes
-deployments of Neo4j instances, checkout xref:addition/kubernetes/index.adoc[this section]
\ No newline at end of file
+deployments of Neo4j instances, checkout xref:addition/kubernetes/index.adoc[this section]
```
This section lists some of the methods to set up agents to monitor containerized Neo4j instances in standalone and Docker compose deployments. 
For Kubernetes deployments of Neo4j instances, see xref:addition/kubernetes/index.adoc[this section] for more information.

AlexicaWright commented on modules/ROOT/pages/addition/instance-requirements.adoc:
```diff
@@ -17,6 +17,23 @@ This setting acts as an lower bound for any completed query to appear in the log
+[[server_log_config]]
+=== Finding out the location of server log configuration (server-logs.xml)
+
+The location of server log configuration file is configured in `neo4j.conf` (see documentation on link:https://neo4j.com/docs/operations-manual/current/configuration/configuration-settings/#config_server.logs.config[server.logs.config] configuration setting).
```
```suggestion
The location of server log configuration file is configured in `neo4j.conf` (see documentation on link:https://neo4j.com/docs/operations-manual/current/configuration/configuration-settings/#config_server.logs.config[Operations Manual -> server.logs.config] configuration setting).
```

AlexicaWright commented on modules/ROOT/pages/addition/instance-requirements.adoc:
```diff
@@ -17,6 +17,23 @@ This setting acts as an lower bound for any completed query to appear in the log
+
+[IMPORTANT]
+====
+Check that the server log configuration exists, otherwise NOM agent will not be able to add query log collection configuration to it.
```
```suggestion
Make sure that the server log configuration exists, otherwise NOM agent will not be able to add query log collection configuration to it.
```

AlexicaWright commented on modules/ROOT/pages/addition/kubernetes/sidecar.adoc:
```diff
@@ -5,7 +5,7 @@ An agent can simply be run in a lightweight linux based container such as `alpin
 [WARNING]
 ====
-An agent Sidecar container can not access all resources that Neo4j pod uses and the host metrics of the cluster node. For example, for disk usage metrics, agent needs access to file system or peristent volumes that Neo4j pod uses. These need to be mounted to agent Daemon pod. Agent sidecar can only report host metrics of it's own contained environment which is not a complete view of reosurce usag ...
+An agent Sidecar container can not access all resources that Neo4j pod uses and the host metrics of the cluster node. For example, for disk usage metrics, agent needs access to file system or persistent volumes that Neo4j pod uses. These need to be mounted to agent Daemon pod. Agent sidecar can only report host metrics of it's own contained environment which is not a complete view of reosurce usa ...
```
```suggestion
An agent Sidecar container cannot access all resources that Neo4j pod uses and the host-metrics of the cluster node. 
For example, for disk usage metrics, the agent needs access to the file system or persistent volumes that Neo4j pod uses. 
These need to be mounted to the agent Daemon pod. 
The agent Sidecar can only report host metrics of its own contained environment which is not a complete view of resource usage of the cluster node.
```

AlexicaWright reviewed (commented):
Some editorial comments.

alexnb committed:
Apply suggestions from code review

Co-authored-by: Jessica Wright <49636617+AlexicaWright@users.noreply.github.com>

alexnb committed:
Apply suggestions from code review

Co-authored-by: Jessica Wright <49636617+AlexicaWright@users.noreply.github.com>

alexnb committed:
editorial changes as suggested
